frontend/src/components/AssetsTable.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ReactTable from 'react-table';
+
+const AssetsTable = (props) => {
+  const { assets } = props;
+
+  const trainInfoKeys = assets.map((asset) => Object.keys(asset.train_info)).flatten();
```

Comment:
You should use `flat` instead of `flatten`.
`flatten` method is provided by babel-polyfill. But `flat` method is provided as browser native.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat